### PR TITLE
Speed up e2e-tests locally by skipping environment setup

### PR DESCRIPTION
**Pull request by noklam**
## Description

While e2e-tests aim to simulate a fresh setup, it takes a lot of time to run the test because 95% of the time the tests is building up a new `venv` and `pip install` which is unlikely the problem.

I introduce a `BEHAVE_LOCAL_ENV` flag to skip this setup and additionally a `e2e-tests-fast` command in MAKEFILE. When developer want to quickly run some test, they can choose to enable the flag.

However, the CI should always run the full test, this is mainly for local development.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by noklam** [comment]
I did a quick test that it doesn't work very well on GitPod yet, locally it works fine. It seems to be a problem of the path of the virtualenv on GitPod.

File: features/environment.py [hunk 1]
```diff
@@ -93,27 +94,39 @@ def _create_tmp_dir() -> Path:
...
+                "pip>=21.2,<23.2",
+            ],
+            env=context.env,
+        )
+        call([context.python, "-m", "pip", "install", "-e", "."], env=context.env)
+        return context
 
 
 def _install_project_requirements(context):
+    return
```
Comment (ankatiyar) [inline features/environment.py:129]: is this intentional?

Comment (noklam) [reply features/environment.py:129]: They are not!

**Review by ankatiyar — APPROVED** [review]
Thanks @noklam, this is pretty cool! :D

File: Makefile [hunk 2]
```diff
@@ -20,6 +20,10 @@ show-coverage:
 e2e-tests:
 	behave --tags=-skip
 
+e2e-tests-fast: export BEHAVE_LOCAL_ENV=TRUE
+e2e-tests-fast:
+	behave --tags=-skip --no-capture
```
Comment (ankatiyar) [inline Makefile:25]: We can also get rid of the --tags=-skip part there's no skip tagged tests anymore

**Review by SajidAlamQB — APPROVED** [review]
Nice, super handy for development! Thank you @noklam! 👍

File: Makefile (hunk 2 above)
Comment (astrojuanlu) [inline Makefile:25]: Can't this be

```suggestion
e2e-tests-fast:
	BEHAVE_LOCAL_ENV=TRUE behave --tags=-skip --no-capture
```

?

Comment (noklam) [reply Makefile:25]: I tried this but it doesn't work. I am not an expert of Makefile so I am happy with change if you found a fix.

image

**Pull request merged**